static_frame/core/frame.py

```diff
@@ -2098,6 +2098,88 @@ def from_parquet(cls,
                 name=name
                 )
 
+    @staticmethod
+    @doc_inject(selector='constructor_frame')
+    def from_msgpack(msgpack_data: bin):
+        '''Frame constructor from an in-memory binary object formatted as a msgpack.
+
+        Args:
+            msgpack_data: A binary msgpack object, encoding a object as produced from to_msgpack()
+
+        Returns:
+            :obj
+        '''
+        import msgpack
+        import msgpack_numpy
+        def decode(obj, chain=msgpack_numpy.decode):
+            if b'sf' in obj:
+                clsname = obj[b'sf']
+                cls = globals()[clsname]
+                if issubclass(cls, Frame):
+                    '''
+                    Classes:
+                        *ContainerBase
+                        *ContainerOperand
+                        Frame
+                        FrameGO
+                    '''
+                    blocks = unpackb(obj[b'blocks']) #recurse unpackb
+                    return cls(
+                            blocks,
+                            name=obj[b'name'],
+                            index=unpackb(obj[b'index']), #recurse unpackb
+                            columns=unpackb(obj[b'columns']), #recurse unpackb
+                            own_data=True)
+                elif issubclass(cls, IndexHierarchy):
+                    '''
+                    Classes:
+                        *ContainerBase
+                        *ContainerOperand
+                        *IndexBase            #CONFLICT: IndexBase must come after IndexHierarchy.
+                        IndexHierarchy
+                        IndexHierarchyGO
+                    '''
+                    blocks = unpackb(obj[b'blocks']) #recurse unpackb
+                    return cls._from_type_blocks(
+                            blocks=blocks,
+                            name=obj[b'name'],
+                            own_blocks=True)
+                elif issubclass(cls, IndexBase):
```

Comment:
Changed from IndexBase to Index. 

I think I understand the problem fully now, and the intention of the IndexBase class. Since as a base class IndexBase is never called directly, we don't need to worry about a direct instance of it being an attribute of the data frame. Good thinking, keeps things clean.

Reply:
also updated requirements files